src/components/Bookie.jsx

```diff
@@ -137,53 +150,50 @@ class Bookie extends Component {
         { startIsFocused: true, startPicking: true, endPicking: false }
         :
         { endIsFocused: true, endPicking: true, startPicking: false };
+
+      // // debug version
+      // diff = isStartNotEnd ?
+      //   { startPicking: true, endPicking: false }
+      //   :
+      //   { endPicking: true, startPicking: false };
+
     } else if ('blur' === eventName) {
-      diff = isStartNotEnd ?
-        {
-          startIsFocused: false,
-          startVal: value
-        }
-        :
-        {
-          endIsFocused: false,
-          endVal: value
-        };
+      diff = isStartNotEnd ? { startIsFocused: false } : { endIsFocused: false };
+      diff.rangeEndValEntered = value;
     }
 
     this.negotiateStateDiff(diff);
-
-    console.log(isStartNotEnd, eventName);
   }
 
   render() {
-    const s = this.state;
-    // const isWaiting = (s.loading || !s.initialized);
-    console.log(s);
-
     const {
-      startPicking, endPicking, startVal, endVal, startIsFocused, endIsFocused
+      forbidDayBack, forbidHourBack,
+      daysFrame, hoursFrame, qMinutesFrame,
+      startPicking, endPicking, startValStr, endValStr, startIsFocused, endIsFocused
     } = this.state;
 
     return <div className="bookie-container">
       <PickerItemList className="pick-day"
-        items={s.daysFrame}
+        items={daysFrame}
         onClick={(item, index) => { this.clickDay(item); } }
         wrapWithArrows
+        forbidBack={forbidDayBack}
         />
 
       <PickerItemList className="pick-minutes"
-        items={s.qMinutesFrame}
+        items={qMinutesFrame}
         onClick={(item, index) => { this.clickQuarter(index); } }
         />
 
       <PickerItemList className="pick-hour"
-        items={s.hoursFrame}
+        items={hoursFrame}
         onClick={(item, index) => { this.clickHour(item); } }
         wrapWithArrows
+        forbidBack={forbidHourBack}
         />
 
       <PickerDateRange
-        data={{ startPicking, endPicking, startVal, endVal, startIsFocused, endIsFocused }}
+        data={{ startPicking, endPicking, startValStr, endValStr, startIsFocused, endIsFocused }}
```

Comment:
Props should be passed into component individualy to have control over prop types and count